contracts/pcv/maple/MaplePCVDeposit.sol

```diff
@@ -0,0 +1,354 @@
+            IMaplePool(maplePool).recognizableLossesOf(address(this));
+    }
+
+    /// @notice return the underlying token denomination for this deposit
+    function balanceReportedIn() external view returns (address) {
+        return USDC;
+    }
+
+    /// @notice set the pcv oracle address
+    /// @param _pcvOracle new pcv oracle to reference
+    function setPCVOracle(address _pcvOracle) external onlyGovernor {
+        address oldOracle = pcvOracle;
+        pcvOracle = _pcvOracle;
+
+        _recordPNL();
+
+        IPCVOracle(pcvOracle).updateIlliquidBalance(
+            lastRecordedBalance.toInt256()
+        );
+
+        emit PCVOracleUpdated(oldOracle, _pcvOracle);
+    }
+
+    /// ---------- Happy Path APIs ----------
+
+    /// @notice deposit PCV into Maple.
+    /// deposits are subject to up to 30 days of lockup,
+    /// weighted by the amount deposited and the deposit time.
+    /// No op if 0 token balance.
+    /// Deposits are then immediately staked to accrue MPL rewards.
+    /// Only pcv controller can deposit, as this contract would be vulnerable
+    /// to donation / griefing attacks if anyone could call deposit and extend lockup time.
+    function deposit() external onlyPCVController {
+        uint256 amount = IERC20(USDC).balanceOf(address(this));
+        if (amount == 0) {
+            /// no op to prevent wasted gas
+            return;
+        }
+
+        int256 startingRecordedBalance = lastRecordedBalance.toInt256();
+
+        /// ------ Effects ------
+
+        /// compute profit from interest accrued and emit an event
+        /// if any profits or losses are realized
+        _recordPNL();
+
+        lastRecordedBalance += amount;
+
+        /// ------ Interactions ------
+
+        /// pool deposit
+        IERC20(USDC).approve(maplePool, amount);
+        IMaplePool(maplePool).deposit(amount);
+
+        /// stake pool FDT for MPL rewards
+        uint256 scaledDepositAmount = amount * SCALING_FACTOR;
+        IMaplePool(maplePool).increaseCustodyAllowance(
+            mapleRewards,
+            scaledDepositAmount
+        );
+        IMapleRewards(mapleRewards).stake(scaledDepositAmount);
+
+        int256 endingRecordedBalance = lastRecordedBalance.toInt256();
+
+        if (pcvOracle != address(0)) {
+            IPCVOracle(pcvOracle).updateIlliquidBalance(
+                endingRecordedBalance - startingRecordedBalance
+            );
+        }
+
+        emit Deposit(msg.sender, amount);
+    }
+
+    /// @notice function to start the cooldown process to withdraw
+    /// 1. lp lockup on deposit --> 90 days locked up and can't withdraw
+    /// 2. cool down period, call intend to withdraw -->
+    ///   must wait 10 days before withdraw after calling intend to withdraw function
+    /// 3. after cool down and past the lockup period,
+    ///    have 2 days to withdraw before cool down period restarts.
+    function signalIntentToWithdraw() external onlyPCVController {
```

Comment:
Let's make this a generic function for all illiquid deposits, and make an illiquid pcv deposit interface so our next illiquid venue has a standard interface handle withdraws if any signaling is needed.

function could be named `startUnlock` or `startWithdraw`.